torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -99,7 +99,8 @@ struct TORCH_API AccumulateGrad : public Node {
       if (!GradMode::is_enabled() &&
           !new_grad.is_sparse() &&
           new_grad.use_count() <= num_expected_refs &&
-          utils::obeys_layout_contract(new_grad, variable)) {
+          ((!new_grad.is_mkldnn() &&utils::obeys_layout_contract(new_grad, variable))
+           || new_grad.is_mkldnn())){
```

Comment:
Does this mean that the layout contract is not respected for mkldnn Tensors? Should we update the comment describing it above to reflect that?

Reply:
Yes, MKLDNN tensor is a opaque tensor which is not obey the layout contract.